Empire - The Last Battle/Assets/Unity/Scripts/MonoBehaviours/Armoury.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class PurchasableItem : ScriptableObject
+{
+    public int Cost { get; set; }
+}
+
+public class PurchasableUnit : PurchasableItem
+{
+    //Change to UnitType once units are in
+    public int Id { get; set; }
+
+    //This is 0-4, and depends on how many 
+    //castle pieces player has
+    public int PurchaseLevel { get; set; }
+}
+
+public class PurchasbleCard : PurchasableItem
+{
+    public int Id { get; set; }
+}
+
+public class PurchasbleCastlePiece : PurchasableItem
+{
+    public int Id { get; set; }
+}
+
+public class Armoury : MonoBehaviour
+{
+    public List<PurchasableUnit> PurchasbleUnits { get; set; }
+    public List<PurchasbleCard> PurchasbleCards { get; set; }
+    public List<PurchasbleCastlePiece> PurchasableCastlePieces { get; set; }
+
+    public delegate void PurchasedItemCallback(PurchasableItem purchasedItem);
+    public event PurchasedItemCallback OnPurchasedItem = delegate { };
+
+    void Initialise()
+    {
+        PurchasbleUnits = new List<PurchasableUnit>
+        {
+            //Add units here when unit class is in
+        };
+
+        PurchasbleCards = new List<PurchasbleCard>
+        {
+            //Add cards here when cards class is in 
+        };
+
+        PurchasableCastlePieces = new List<PurchasbleCastlePiece>
+        {
+            new PurchasbleCastlePiece(){ Id = 0, Cost = 2000 },
```

Comment:
All these would be created as scriptableobject files just like the tiles, then dragged into the inspector array, so they wouldn't need to be initialised. 

The old way with the dictionary would have to be, since dictionaries cannot be edited in the inspector.